### ISSUE TITLE
Injectables are not resolved properly inside angular element - web component

### ISSUE THREAD

**Issue by vinayk406** [angular/angular#45254]
### Which @angular/* package(s) are the source of the bug?

elements

### Is this a regression?

No

### Description

When an application is bundled as a webcomponent using angular elements, injectables within the library module are resolved using NodeInjector whereas they should be resolved using closest ModuleInjector. 

---
**working solution - without using angular elements**

**expected output:**

- `TranslateService` injected in `TestModule` and `TestComponent` are same in this case.

<img width="951" alt="Screenshot 2022-03-03 at 11 39 46 PM" src="https://user-images.githubusercontent.com/5180606/156625836-f79fddc2-7d5a-41e3-b3b5-d4da15a77095.png">

stackblitz: https://stackblitz.com/edit/github-tovn9d

---
**Issue with angular elements**

- `TranslateService` injected in `TestModule` and `TestComponent` are **NOT same** in this case. 

<img width="951" alt="Screenshot 2022-03-03 at 11 47 08 PM" src="https://user-images.githubusercontent.com/5180606/156626953-854388cc-356a-4f7c-ba25-809abbdc701e.png">

stackblitz: https://stackblitz.com/edit/github-tovn9d-rvb918

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/github-tovn9d-rvb918

### Please provide the environment you discovered this bug in (run `ng version`)

```true
"@angular/animations": "~13.2.0",
"@angular/common": "~13.2.0",
"@angular/compiler": "~13.2.0",
"@angular/core": "~13.2.0",
"@angular/elements": "^13.2.5",
"@angular/forms": "~13.2.0",
"@angular/platform-browser": "~13.2.0",
"@angular/platform-browser-dynamic": "~13.2.0",
"@angular/router": "~13.2.0",
"@ngx-translate/core": "^14.0.0",
"@nrwl/angular": "13.8.4",
"rxjs": "~7.4.0",
```

**Comment by JoostK** [comment]
This seems to be a duplicate of #37441, but the repro is extremely hard to follow and won't run for me on StackBlitz. Can you try if https://github.com/angular/angular/issues/37441#issuecomment-647054694 makes a difference?

**Comment by vinayk406** [comment]
@JoostK workaround mentioned in the comment is working for me. 
Thanks.

**Comment by JoostK** [comment]
Thanks for confirming, then it's indeed a duplicate!
